modules/apm/src/main/java/org/elasticsearch/telemetry/apm/APMMeterRegistry.java

```diff
@@ -127,6 +133,35 @@ public LongCounter registerLongCounter(String name, String description, String u
         }
     }
 
+    @Override
+    public LongAsyncCounter registerLongAsyncCounter(String name, String description, String unit, Supplier<LongWithAttributes> observer) {
+        try (ReleasableLock lock = registerLock.acquire()) {
+            return longAsynchronousCounters.register(new LongAsyncCounterAdapter(meter, name, description, unit, observer));
+        }
+    }
+
+    @Override
+    public LongAsyncCounter getLongAsyncCounter(String name) {
```

Comment:
Do we need this method? I guess the only thing you can do is to call close() on it - is it useful?

Reply:
fair point. I created it for consistency only. I think we should revisit others too with callback style. will create an issue